ActivityHistoryApi/V1/Infrastructure/ActivityHistoryDB.cs

```diff
@@ -0,0 +1,40 @@
+using ActivityHistoryApi.V1.Domain;
+using Amazon.DynamoDBv2.DataModel;
+using Hackney.Core.DynamoDb.Converters;
+using System;
+using Type = ActivityHistoryApi.V1.Domain.Type;
+
+namespace ActivityHistoryApi.V1.Infrastructure
+{
+    [DynamoDBTable("ActivityHistory", LowerCamelCaseProperties = true)]
+    public class ActivityHistoryDB
+    {
+        [DynamoDBRangeKey]
+        public Guid Id { get; set; }
+
+        [DynamoDBHashKey]
+        public Guid TargetId { get; set; }
+
+
+        [DynamoDBProperty(Converter = typeof(DynamoDbEnumConverter<Type>))]
+        public Type Type { get; set; }
+
+        [DynamoDBProperty(Converter = typeof(DynamoDbEnumConverter<TargetType>))]
+        public TargetType TargetType { get; set; }
+
+        [DynamoDBProperty(Converter = typeof(DynamoDbDateTimeConverter))]
+        public DateTime CreatedAt { get; set; }
+
+        public int TimetoLiveForRecordInDays { get; set; }
+
+        [DynamoDBProperty(Converter = typeof(DynamoDbObjectConverter<OldData>))]
+        public OldData OldData { get; set; }
+
+        [DynamoDBProperty(Converter = typeof(DynamoDbObjectConverter<NewData>))]
+        public NewData NewData { get; set; }
```

Comment:
The old and new data fields don't need typed model classes.
As they have dynamic data you should just use `Dictionary<string,object>` for each of them, and when serialised it will be the correct format. (They might not even needed a converter specified but I would verify that as the AWS SDK is quirky like that.)